src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ArgumentProvider.cs

```diff
@@ -0,0 +1,285 @@
+            : base(defaultNamespace, sourceInputModel)
+        {
+            DeclarationModifiers = TypeSignatureModifiers.Internal | TypeSignatureModifiers.Static;
+            _nameParamRef = new ParameterReference(_nameParam);
+            _nullableT = _t.WithNullable(true);
+        }
+
+        protected override string DefaultName => "Argument";
+
+        private MethodSignature GetSignature(
+            string name,
+            IReadOnlyList<Parameter> parameters,
+            IReadOnlyList<CSharpType>? genericArguments = null,
+            IReadOnlyDictionary<CSharpType, FormattableString>? genericArgumentConstraints = null,
+            CSharpType? returnType = null)
+        {
+            return new MethodSignature(
+                name,
+                null,
+                null,
+                MethodSignatureModifiers.Static | MethodSignatureModifiers.Public,
+                returnType,
+                null,
+                parameters,
+                GenericArguments: genericArguments,
+                GenericParameterConstraints: genericArgumentConstraints);
+        }
+
+        protected override IEnumerable<Method> BuildMethods()
+        {
+            yield return BuildAssertNotNull();
+            yield return BuildAssertNotNullStruct();
+            yield return BuildAssertNotNullOrEmptyCollection();
+            yield return BuildAssertNotNullOrEmptyString();
+            yield return BuildAssertNotNullOrWhiteSpace();
+            yield return BuildAssertNotDefault();
+            yield return BuildAssertInRange();
+            yield return BuildAssertEnumDefined();
+            yield return BuildCheckNotNull();
+            yield return BuildCheckNotNullOrEmptyString();
+            yield return BuildAssertNull();
+        }
+
+        private Method BuildAssertNull()
+        {
+            var valueParam = new Parameter("value", null, _t, null, ValidationType.None, null);
+            var messageParam = new Parameter("message", null, typeof(string), Constant.Default(new CSharpType(typeof(string), true)), ValidationType.None, null);
+            var signature = GetSignature("AssertNull", new[] { valueParam, _nameParam, messageParam }, new[] { _t });
+            return new Method(signature, new MethodBodyStatement[]
+            {
+                new IfStatement(NotEqual(new ParameterReference(valueParam), Null))
+                {
+                    ThrowArgumentException(NullCoalescing(new ParameterReference(messageParam), Literal("Value must be null.")))
+                }
+            });
+        }
+
+        private Method BuildCheckNotNullOrEmptyString()
+        {
+            var valueParam = new Parameter("value", null, typeof(string), null, ValidationType.None, null);
+            var signature = GetSignature("CheckNotNull", new[] { valueParam, _nameParam }, returnType: typeof(string));
+            var valueParamRef = new ParameterReference(valueParam);
```

Comment:
I would like to name this expression with the same name it is written, for instance here we could name this as `value`
```suggestion
            var value = new ParameterReference(valueParam);
```
and then when we are building the statements, the statements would look like the code we write.
```csharp
var value = new ParameterReference(valueParam);
var name = new ParameterReference(_nameParam);
return new Method(signature, new MethodBodyStatement[]
{
    AssertNotNullOrEmpty(value, name),
    Return(value)
});
```